plugins/modules/pnp_intent.py

```diff
@@ -566,29 +581,31 @@ def get_have(self):
                     params=self.want.get("image_params"),
                 )
                 image_list = image_response.get("response")
-                self.log(str(image_response))
+                self.log("Image details obtained from the API 'get software image details' are {0}".format(str(image_response)), "DEBUG")
 
                 # check if project has templates or not
                 template_list = self.dnac_apply['exec'](
                     family="configuration_templates",
                     function='gets_the_templates_available',
                     params={"project_names": self.want.get("project_name")},
                 )
-                self.log(str(template_list))
+                self.log("Templates list under the project {0} is {1}".format(self.want.get("project_name"), str(template_list)), "DEBUG")
 
                 dev_details_response = self.dnac_apply['exec'](
                     family="device_onboarding_pnp",
                     function="get_device_by_id",
                     params={"id": device_response[0].get("id")}
                 )
-
+                self.log("Device details of the existing device after calling the API 'get device by id' are {0}".format(str(dev_details_response)), "DEBUG")
                 install_mode = dev_details_response.get("deviceInfo").get("mode")
+                self.log("Installation mode of the device with the serial no. {0} is {1}".format(self.want.get("serial_number"), install_mode), "INFO")
```

Comment:
self.log("Installation mode of the device with serial number '{0}': {1}".format(self.want.get("serial_number"), install_mode), "INFO")

Reply:
Resolved